File: app/src/main/java/doit/study/droid/data/source/local/entities/Tag.java
package doit.study.droid.data.source.local.entities;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Ignore;
import android.arch.persistence.room.PrimaryKey;

import java.util.List;

import doit.study.droid.data.source.local.Converters;

@Entity(tableName = "tags")
public class Tag {
    @PrimaryKey(autoGenerate = true)
    public int id;
    public final String text;
    public final int questionId;

    public Tag(int id, String text, int questionId) {
        this.id = id;
        this.text = text;
        this.questionId = questionId;
    }

    @Ignore
    public Tag(List<String> text, int questionId) {
        this.text = Converters.listToString(text);
        this.questionId = questionId;
    }
}

File: app/src/main/java/doit/study/droid/data/local/entity/Tag.kt
package doit.study.droid.data.local.entity

import androidx.room.Entity
import androidx.room.PrimaryKey
import doit.study.droid.topic.TopicItem

@Entity
data class Tag(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    val name: String,
        // Read/Write
    val selected: Boolean = false
)


internal fun Tag.toTagItem(): TopicItem {
    TODO("not yet")
}

internal fun TopicItem.toTag(selected: Boolean): Tag {
    return Tag(
            id = this.id,
            name = this.name,
            selected = selected
    )
}